dev-lisp/roswell/roswell-19.3.10.97.ebuild

```diff
@@ -14,6 +14,8 @@ LICENSE="MIT"
 
 KEYWORDS="~amd64 ~x86"
 
+RDEPEND="!net-libs/librouteros"
```

Comment:
No no, THIS is a soft blocker and will break librouteros after emerging roswell ;)

Put a hard blocker instead 
`RDEPEND="!!net-libs/librouteros"`

Please test that it works correctly, ie after installing roswell try to emerge librouteros. Also you can use `Closes:` instead of `Bug:` in your commit message to automatically close the bug when this gets merged.

Reply:
Hmm...Actually I tried to emerge `net-libs/librouteros` and it says:

```
[ebuild  N     ] net-libs/librouteros-1.1.2  USE="-debug -static-libs"
[blocks B      ] net-libs/librouteros ("net-libs/librouteros" is blocking dev-lisp/roswell-19.3.10.97)

 * Error: The above package list contains packages which cannot be
 * installed at the same time on the same system.

  (net-libs/librouteros-1.1.2:0/0::gentoo, ebuild scheduled for merge) pulled in by
    librouteros

  (dev-lisp/roswell-19.3.10.97:0/0::local, installed) pulled in by
    dev-lisp/roswell required by @selected
```

Is this not enough?

Reply:
Soft blocker allows the installation of these packages simultaneously, and then the package manager will remove the "blocked" package, while hard block will stop the user and make them realize what's happening and do the decision before any action is taken. 
```
Calculating dependencies... done!
[ebuild  N     ] net-libs/librouteros-1.1.2::gentoo  USE="-debug -static-libs" 230 KiB

Calculating dependencies... done!
[ebuild  N     ] dev-lisp/roswell-19.3.10.97::testworld  0 KiB
[uninstall     ] net-libs/librouteros-1.1.2::gentoo  USE="-debug -static-libs" 
[blocks b      ] net-libs/librouteros ("net-libs/librouteros" is blocking dev-lisp/roswell-19.3.10.97)

Total: 1 package (1 new, 1 uninstall), Size of downloads: 0 KiB
Conflict: 1 block

Would you like to merge these packages? [Yes/No] y
>>> Verifying ebuild manifests
>>> Emerging (1 of 1) dev-lisp/roswell-19.3.10.97::testworld
>>> Installing (1 of 1) dev-lisp/roswell-19.3.10.97::testworld
>>> Uninstalling net-libs/librouteros-1.1.2::gentoo
>>> Jobs: 1 of 1 complete                           Load avg: 7.24, 7.74, 7.81
>>> Auto-cleaning packages...

Calculating dependencies... done!
[ebuild  N     ] net-libs/librouteros-1.1.2::gentoo  USE="-debug -static-libs" 0 KiB
[uninstall     ] dev-lisp/roswell-19.3.10.97::testworld 
[blocks b      ] net-libs/librouteros ("net-libs/librouteros" is blocking dev-lisp/roswell-19.3.10.97)

Total: 1 package (1 new, 1 uninstall), Size of downloads: 0 KiB
Conflict: 1 block

Would you like to merge these packages? [Yes/No] y
>>> Verifying ebuild manifests
>>> Emerging (1 of 1) net-libs/librouteros-1.1.2::gentoo
>>> Installing (1 of 1) net-libs/librouteros-1.1.2::gentoo
>>> Uninstalling dev-lisp/roswell-19.3.10.97::testworld
>>> Jobs: 1 of 1 complete                           Load avg: 8.02, 7.84, 7.84
>>> Auto-cleaning packages...

Calculating dependencies... done!
[ebuild  N     ] dev-lisp/roswell-19.3.10.97::testworld  0 KiB
[blocks B      ] net-libs/librouteros ("net-libs/librouteros" is hard blocking dev-lisp/roswell-19.3.10.97)

Total: 1 package (1 new), Size of downloads: 0 KiB
Conflict: 1 block (1 unsatisfied)

 * Error: The above package list contains packages which cannot be
 * installed at the same time on the same system.

  (dev-lisp/roswell-19.3.10.97:0/0::testworld, ebuild scheduled for merge) pulled in by
    roswell
```
I'd say it's up to what you prefer, just that the softblocker might go unnoticed and apparently in your case since it's recorded in world file it might work better.
Let me know which you prefer.

Reply:
I think the hard block should be better since there is a binary conflict, which is "hard" enough.

By the way should I add another commit to fix this, or force overwrite the current one?